runtime/CSharp/src/Atn/LexerActionExecutor.cs

```diff
@@ -98,9 +99,11 @@ public static Antlr4.Runtime.Atn.LexerActionExecutor Append(Antlr4.Runtime.Atn.L
             {
                 return new Antlr4.Runtime.Atn.LexerActionExecutor(new ILexerAction[] { lexerAction });
             }
-            ILexerAction[] lexerActions = Arrays.CopyOf(lexerActionExecutor.lexerActions, lexerActionExecutor.lexerActions.Length + 1);
-            lexerActions[lexerActions.Length - 1] = lexerAction;
-            return new Antlr4.Runtime.Atn.LexerActionExecutor(lexerActions);
+            var lexerActions = lexerActionExecutor.lexerActions;
+            ILexerAction[] newLexerActions  = new ILexerAction[lexerActions.Length + 1];
```

Comment:
Please use Array.Resize for consistency

Reply:
I'm not sure it's correct code. Should `Append` method change internal state of instance if consider it returns another instance? It looks like unwanted side effect.

Also, there is no state changing in Java runtime, only copying: https://github.com/antlr/antlr4/blob/b5648f405b746d87ec559132ad1939fb6d402053/runtime/Java/src/org/antlr/v4/runtime/atn/LexerActionExecutor.java#L72 The code is just incorrect.

Reply:
ok